src/main/java/com/sysc4806app/server/ProductController.java

```diff
@@ -33,11 +35,24 @@ public String productList(Model model) {
     }
 
     @GetMapping(path="/product/{id}")
-    public String requestProduct(@PathVariable("id") long id, Model model) {
+    public String requestProduct(@PathVariable("id") long id, @RequestParam(value = "sort", required = false) String sort, Model model) {
         Product product = productRepo.findById(id);
         double rating = productService.calculateRating(id);
+        List<Review> reviews;
+
+        if (sort == null) {
+            reviews = reviewRepo.findByProduct(product);
+        } else {
+            Sort.Direction dir = Sort.Direction.DESC;
+            if (sort.startsWith("-")) {
```

Comment:
I feel like we should allow a + at the start without it breaking. Also shouldn't a '-' make it descending, not the other way around?

Reply:
The '-' is arbitrary just to reverse the default order, and I wanted the default to show highest values first because it is convenient for sorting on ratings, etc.

Reply:
Also plus sign has a specific meaning in url